frontend/src/metabase/visualizations/visualizations/Scalar.jsx

```diff
@@ -29,6 +29,27 @@ function legacyScalarSettingsToFormatOptions(settings) {
     .value();
 }
 
+const PADDING = 32;
+const TITLE_ICON_SIZE = 10;
+const SCALAR_TITLE_LINE_HEIGHT = 23;
+
+const TITLE_2_LINES_HEIGHT_THRESHOLD = 120; // determined empirically
+
+const getTitleLinesCount = height =>
+  height > TITLE_2_LINES_HEIGHT_THRESHOLD ? 2 : 1;
+
+const getTitleHeight = ({ isDashboard, showSmallTitle, titleLinesCount }) => {
```

Comment:
wdyt about moving these utilities to a different (utility) file? I'd also recommend taking a look at how static viz charts (e.g. RowChart) are implemented where there are utility functions that return props for "view" components. Here we could have a function that returns values for props for inner components so there won't be any computation in the component; and we could also test these utility functions

Reply:
With pleasure :rocket: 

07038432ac74e5f7bad50a7e957dbfa66de149b5 + d502456f3743789e72c5179fee5896180902c5bd

Reply:
> and we could also test these utility functions

I didn't add any tests for them. There are 2 types of functions that I managed to extract:
- trivial ones involving arithmetic, which I think don't deserve tests (TS seems enough of a check)
- those using `measureText` - it'd be awkward to unit-test them, as we'd have to mock `measureText` - not sure if this whole endeavor would make sense - please let me know if you think so

Reply:
Added unit tests for most functions: bd8bb0e7daa8a192e0f114dde071d0c437b52bf4 + 408df899c245c0a396fcadfb8d20f7a9e0405e4d